src/lib/services/winners-service.ts

```diff
@@ -0,0 +1,50 @@
+import { ResultRow } from '../../models/database';
+import { PockyDB } from '../database/db-interfaces';
+import TableHelper from "../parsers/tableHelper";
+import __logger from "../logger";
+import { Receiver } from "../../models/receiver";
+
+export interface IWinnersService {
+	returnWinnersResponse(): Promise<string>
+}
+
+export default class WinnersService implements IWinnersService {
+	readonly cannotDisplayResults: string = 'Error encountered; cannot display winners.';
+	database: PockyDB;
+
+	constructor(database: PockyDB) {
+		this.database = database;
+	}
+
+	async returnWinnersResponse() : Promise<string> {
+		const data: ResultRow[] = await this.database.returnWinners();
+
+		let winners: Receiver[] = TableHelper.mapResults(data);
+		let columnWidths = TableHelper.getReceiverColumnWidths(winners);
+
+		// define table heading
+		let winnersTable = TableHelper.padString('Receiver', columnWidths.receiver) + ' | ' + TableHelper.padString('Sender', columnWidths.sender) + ' | Comments\n';
+		winnersTable += 'Total'.padEnd(columnWidths.receiver) + ' | ' + ' '.padEnd(columnWidths.sender) + ' | \n';
+		winnersTable += ''.padEnd(columnWidths.receiver, '-') + '-+-' + ''.padEnd(columnWidths.sender, '-') + '-+-' + ''.padEnd(columnWidths.comment, '-') + '\n';
+
+		__logger.debug('Building winners table');
+
+		// put in table data
+		winners.forEach((winner: Receiver) => {
+			winnersTable += winner.person.toString().padEnd(columnWidths.receiver) + ' | ' + ''.padEnd(columnWidths.sender) + ' | \n';
+			let firstPeg = true;
+			let pegCount = winner.pegs.length;
+			winner.pegs.forEach((peg) => {
+				if (firstPeg) {
+					winnersTable += pegCount.toString().padEnd(columnWidths.receiver) + ' | ' + peg.sender.toString().padEnd(columnWidths.sender) + ' | ' + peg.comment + '\n';
+					firstPeg = false;
+				} else {
+					winnersTable += ''.padEnd(columnWidths.receiver) + ' | ' + peg.sender.toString().padEnd(columnWidths.sender) + ' | ' + peg.comment + '\n';
+				}
+			});
+		});
+
+		__logger.information(`########### Winners table:\n\n${winnersTable}`);
```

Comment:
logs with newlines won't appear as a single log in kibana, but I guess the winners table has newlines in it soo...

Reply:
Yeah that's one of the things I was working on in my fix-logging branch.

Reply:
Ignoring this as it's part of the fix logging